base/loading.jl

```diff
@@ -2,6 +2,28 @@
 
 # Base.require is the implementation for the `import` statement
 
+# Generic case-sensitive version of isfile
+function isfile_casesensitive_slow(path)
+    isfile(path) || return false
+    dir, filename = splitdir(path)
+    any(readdir(dir) .== filename)
+end
+
+isfile_casesensitive(path) = isfile_casesensitive_slow(path)
+
+@windows_only function isfile_casesensitive(path)
+    isfile(path) || return false
+    longpath(path) == path
+end
+
+@linux_only isfile_casesensitive(path) = isfile(path)
+
+@osx_only function isfile_casesensitive(path)
+    isfile(path) || return false
+    islink(path) && return isfile_casesensitive_slow(path)
+    realpath(path) == path
```

Comment:
Unless you can find a better solution, I think this should check `basename(realpath(path)) == basename(path)` in case `path` is not a symlink but is in a symlinked directory